desc/compute/utils.py

```diff
@@ -11,45 +12,308 @@
 
     Returns
     -------
-    flag : bool
-        True if the Transforms can compute requested derivatives, False otherwise.
+    data : dict of ndarray
+        Computed quantity and intermediate variables.
 
     """
-    if "R_derivs" not in data_index[key]:
-        R_flag = True
-        Z_flag = True
-    else:
-        R_flag = np.array(
-            [d in R_transform.derivatives.tolist() for d in data_index[key]["R_derivs"]]
-        ).all()
-        Z_flag = np.array(
-            [d in Z_transform.derivatives.tolist() for d in data_index[key]["R_derivs"]]
-        ).all()
-
-    if "L_derivs" not in data_index[key]:
-        L_flag = True
-    else:
-        L_flag = np.array(
-            [d in L_transform.derivatives.tolist() for d in data_index[key]["L_derivs"]]
-        ).all()
+    for name in names:
+        if name not in data_index:
+            raise ValueError("Unrecognized value '{}'.".format(name))
+    allowed_kwargs = {"helicity", "M_booz", "N_booz", "gamma"}
+    bad_kwargs = set(kwargs.keys()).difference(allowed_kwargs)
+    if len(bad_kwargs) > 0:
+        raise ValueError(f"Unrecognized argument(s): {bad_kwargs}")
+
+    for name in names:
+        assert _has_params(name, params), f"Don't have params to compute {name}"
+        assert _has_profiles(name, profiles), f"Don't have profiles to compute {name}"
+        assert _has_transforms(
+            name, transforms
+        ), f"Don't have transforms to compute {name}"
+
+    import desc.compute
+
+    for name in names:
+        fun = getattr(desc.compute, data_index[name]["fun"])
+        data = fun(params, transforms, profiles, data, **kwargs)
+    return data
+
+
+def get_data_deps(*keys):
+    """Get list of data keys needed to compute a given quantity.
+
+    Parameters
+    ----------
+    keys : str
+        Name of the desired quantity from the data index
+
+    Returns
+    -------
+    deps : list of str
+        Names of quantities needed to compute key
+    """
+
+    def _get_deps_1_key(key):
+        if "full_dependencies" in data_index[key]:
+            return data_index[key]["full_dependencies"]["data"]
+        deps = data_index[key]["dependencies"]["data"]
+        if len(deps) == 0:
+            return deps
+        out = deps.copy()
+        for dep in deps:
+            out += _get_deps_1_key(dep)
+        return sorted(list(set(out)))
+
+    out = []
+    for key in keys:
+        out += _get_deps_1_key(key)
+    return sorted(list(set(out)))
+
+
+def get_derivs(*keys):
+    """Get dict of derivative orders needed to compute a given quantity.
+
+    Parameters
+    ----------
+    keys : str
+        Name of the desired quantity from the data index
+
+    Returns
+    -------
+    derivs : dict of list of int
+        Orders of derivatives needed to compute key.
+        Keys for R, Z, L, etc
+    """
+
+    def _get_derivs_1_key(key):
+        if "full_dependencies" in data_index[key]:
```

Comment:
when would this not be the case? since as soon as the compute module is initialized it build the full tree

Reply:
We also call these during the initialization to build the full dependency tree, so need this check. Also if for some reason you added something to the data index at runtime (which should be possible and possibly useful in the future)